src/rustup.rs

```diff
@@ -42,7 +58,12 @@ pub fn installed_toolchains(verbose: bool) -> Result<Vec<String>> {
         .collect())
 }
 
-pub fn available_targets(toolchain: &str, verbose: bool) -> Result<AvailableTargets> {
+pub fn available_targets(
+    // this is explicitly a string and not `QualifiedToolchain`, this is because we use this as a way to ensure that
+    // the toolchain is an official toolchain, if this errors on `is a custom toolchain`, we tell the user to set CROSS_CUSTOM_TOOLCHAIN to handle the logic needed.
+    toolchain: &str,
+    verbose: bool,
+) -> Result<AvailableTargets> {
```

Comment:
this may not be neccessary anymore, I changed the logic for how `QualifiedToolchain` gets output, when I made this comment I simply used the parsed toolchain name, which failed catastrophically when something was not right with the toolchain name